#VRML_SIM R2019a utf8
WorldInfo {
}
Viewpoint {
  orientation 0.6842780440639797 0.6362820047764332 0.35624257018169136 4.911695262803741
  position -4.10078074796391 9.056181131478676 3.1000755420461155
}
TexturedBackground {
}
TexturedBackgroundLight {
}
Floor {
}
Robot {
  translation 0 0.10000000000000002 0
  rotation 0 1 0 1.0471975479999998
  children [
    DEF BODY Shape {
      appearance DEF PURPLE PBRAppearance {
        baseColor 0.6924238956282902 0.16743724727244982 0.9133287556267643
        roughness 1
      }
      geometry Cylinder {
        height 0.08
        radius 0.14
      }
    }
    DEF WH_1 Transform {
      translation 0 0 0.14
      children [
        DEF WHEEL1 HingeJoint {
          jointParameters HingeJointParameters {
            axis 0 0 1
            anchor 0 0 0.02
          }
          device [
            RotationalMotor {
              name "wheel1"
              maxVelocity 30.3687
              maxTorque 0.8473
            }
            PositionSensor {
              name "ps_1"
              resolution 0.003848
            }
          ]
          endPoint Solid {
            translation 0 0 -0.009999999999999998
            rotation 0 0 1 0
            children [
              DEF sr1 HingeJoint {
                jointParameters HingeJointParameters {
                  anchor 0 0.06 0.02
                }
                endPoint Solid {
                  translation 0 0.06 0.02
                  rotation 1 0 0 0
                  children [
                    DEF r1 Transform {
                      rotation 0 0 1 1.5708
                      children [
                        DEF SMALL_WHEEL_SHAPE Shape {
                          appearance PBRAppearance {
                            baseColor 1 0.752941 0.796078
                            roughness 1
                            metalness 0
                          }
                          geometry Cylinder {
                            height 0.03
                            radius 0.012
                          }
                        }
                      ]
                    }
                  ]
                  name "sr1"
                  boundingObject USE r1
                  physics Physics {
                    density -1
                    mass 0.1
                  }
                }
              }
              DEF sr2 HingeJoint {
                jointParameters HingeJointParameters {
                  axis 0.707107 0.707107 0
                  anchor -0.04535 0.04535 0.02
                }
                endPoint Solid {
                  translation -0.04535 0.04535 0.02
                  rotation 0.7071067811865476 0.7071067811865476 0 0
                  children [
                    DEF r2 Transform {
                      rotation 0 0 1 2.27
                      children [
                        USE SMALL_WHEEL_SHAPE
                      ]
                    }
                  ]
                  name "sr2"
                  boundingObject USE r2
                  physics Physics {
                    density -1
                    mass 0.1
                  }
                }
              }
              DEF sr3 HingeJoint {
                jointParameters HingeJointParameters {
                  anchor 0 -0.06 0.02
                }
                endPoint Solid {
                  translation 0 -0.06 0.02
                  rotation 1 0 0 0
                  children [
                    DEF r3 Transform {
                      rotation 0 0 1 1.5708
                      children [
                        USE SMALL_WHEEL_SHAPE
                      ]
                    }
                  ]
                  name "sr3"
                  boundingObject USE r3
                  physics Physics {
                    density -1
                    mass 0.1
                  }
                }
              }
              DEF sr4 HingeJoint {
                jointParameters HingeJointParameters {
                  axis 0.707107 0.707107 0
                  anchor 0.04535 -0.04535 0.02
                }
                endPoint Solid {
                  translation 0.04535 -0.04535 0.02
                  rotation 0.7071067811865476 0.7071067811865476 0 0
                  children [
                    DEF r4 Transform {
                      rotation 0 0 1 2.27
                      children [
                        USE SMALL_WHEEL_SHAPE
                      ]
                    }
                  ]
                  name "sr4"
                  boundingObject USE r4
                  physics Physics {
                    density -1
                    mass 0.1
                  }
                }
              }
              DEF sr5 HingeJoint {
                jointParameters HingeJointParameters {
                  axis 0 1 0
                  anchor -0.06 0 0.02
                }
                endPoint Solid {
                  translation -0.06 0 0.02
                  children [
                    DEF r5 Transform {
                      children [
                        USE SMALL_WHEEL_SHAPE
                      ]
                    }
                  ]
                  name "sr5"
                  boundingObject USE r5
                  physics Physics {
                    density -1
                    mass 0.1
                  }
                }
              }
              DEF sr6 HingeJoint {
                jointParameters HingeJointParameters {
                  axis 0 1 0
                  anchor 0.06 0 0.02
                }
                endPoint Solid {
                  translation 0.06 0 0.02
                  children [
                    DEF r6 Transform {
                      children [
                        USE SMALL_WHEEL_SHAPE
                      ]
                    }
                  ]
                  name "sr6"
                  boundingObject USE r6
                  physics Physics {
                    density -1
                    mass 0.1
                  }
                }
              }
              DEF sr7 HingeJoint {
                jointParameters HingeJointParameters {
                  axis -0.707107 0.707107 0
                  anchor -0.04535 -0.04535 0.02
                }
                endPoint Solid {
                  translation -0.04535 -0.04535 0.02
                  rotation -0.7071067811865476 0.7071067811865476 0 0
                  children [
                    DEF r7 Transform {
                      rotation 0 0 1 0.76
                      children [
                        USE SMALL_WHEEL_SHAPE
                      ]
                    }
                  ]
                  name "sr7"
                  boundingObject USE r7
                  physics Physics {
                    density -1
                    mass 0.1
                  }
                }
              }
              DEF sr8 HingeJoint {
                jointParameters HingeJointParameters {
                  axis -0.707107 0.707107 0
                  anchor 0.04535 0.04535 0.02
                }
                endPoint Solid {
                  translation 0.04535 0.04535 0.02
                  rotation -0.7071067811865476 0.7071067811865476 0 0
                  children [
                    DEF r8 Transform {
                      rotation 0 0 1 0.76
                      children [
                        USE SMALL_WHEEL_SHAPE
                      ]
                    }
                  ]
                  name "sr8"
                  boundingObject USE r8
                  physics Physics {
                    density -1
                    mass 0.1
                  }
                }
              }
              DEF WHEEL_TRANS Transform {
                translation 0 0 0.02
                rotation 1 0 0 1.5708
                children [
                  DEF WHEEL_SHAPE Shape {
                    appearance PBRAppearance {
                      baseColor 0.9333333333333333 0.33600366216525523 0.33600366216525523
                      roughness 1
                    }
                    geometry Cylinder {
                      height 0.02
                      radius 0.06
                      subdivision 20
                    }
                  }
                ]
              }
            ]
            boundingObject USE WHEEL_TRANS
            physics Physics {
            }
          }
        }
      ]
    }
    DEF WH_2 Transform {
      translation 0.13 0 -0.069
      rotation 0 1 0 2.094395096
      children [
        DEF WHEEL1 HingeJoint {
          jointParameters HingeJointParameters {
            axis 0 0 1
            anchor 0 0 0.02
          }
          device [
            RotationalMotor {
              name "wheel2"
              maxVelocity 30.3687
              maxTorque 0.8473
            }
            PositionSensor {
              name "ps_2"
              resolution 0.003848
            }
          ]
          endPoint Solid {
            translation 0 0 -0.009999999999999998
            rotation 0 0 1 0
            children [
              DEF sr1 HingeJoint {
                jointParameters HingeJointParameters {
                  anchor 0 0.06 0.02
                }
                endPoint Solid {
                  translation 0 0.06 0.02
                  rotation 1 0 0 0
                  children [
                    DEF r1 Transform {
                      rotation 0 0 1 1.5708
                      children [
                        DEF SMALL_WHEEL_SHAPE Shape {
                          appearance PBRAppearance {
                            baseColor 1 0.752941 0.796078
                            roughness 1
                            metalness 0
                          }
                          geometry Cylinder {
                            height 0.03
                            radius 0.012
                          }
                        }
                      ]
                    }
                  ]
                  name "sr1"
                  boundingObject USE r1
                  physics Physics {
                    density -1
                    mass 0.1
                  }
                }
              }
              DEF sr2 HingeJoint {
                jointParameters HingeJointParameters {
                  axis 0.707107 0.707107 0
                  anchor -0.04535 0.04535 0.02
                }
                endPoint Solid {
                  translation -0.04535 0.04535 0.02
                  rotation 0.7071067811865476 0.7071067811865476 0 0
                  children [
                    DEF r2 Transform {
                      rotation 0 0 1 2.27
                      children [
                        USE SMALL_WHEEL_SHAPE
                      ]
                    }
                  ]
                  name "sr2"
                  boundingObject USE r2
                  physics Physics {
                    density -1
                    mass 0.1
                  }
                }
              }
              DEF sr3 HingeJoint {
                jointParameters HingeJointParameters {
                  anchor 0 -0.06 0.02
                }
                endPoint Solid {
                  translation 0 -0.06 0.02
                  rotation 1 0 0 0
                  children [
                    DEF r3 Transform {
                      rotation 0 0 1 1.5708
                      children [
                        USE SMALL_WHEEL_SHAPE
                      ]
                    }
                  ]
                  name "sr3"
                  boundingObject USE r3
                  physics Physics {
                    density -1
                    mass 0.1
                  }
                }
              }
              DEF sr4 HingeJoint {
                jointParameters HingeJointParameters {
                  axis 0.707107 0.707107 0
                  anchor 0.04535 -0.04535 0.02
                }
                endPoint Solid {
                  translation 0.04535 -0.04535 0.02
                  rotation 0.7071067811865476 0.7071067811865476 0 0
                  children [
                    DEF r4 Transform {
                      rotation 0 0 1 2.27
                      children [
                        USE SMALL_WHEEL_SHAPE
                      ]
                    }
                  ]
                  name "sr4"
                  boundingObject USE r4
                  physics Physics {
                    density -1
                    mass 0.1
                  }
                }
              }
              DEF sr5 HingeJoint {
                jointParameters HingeJointParameters {
                  axis 0 1 0
                  anchor -0.06 0 0.02
                }
                endPoint Solid {
                  translation -0.06 0 0.02
                  children [
                    DEF r5 Transform {
                      children [
                        USE SMALL_WHEEL_SHAPE
                      ]
                    }
                  ]
                  name "sr5"
                  boundingObject USE r5
                  physics Physics {
                    density -1
                    mass 0.1
                  }
                }
              }
              DEF sr6 HingeJoint {
                jointParameters HingeJointParameters {
                  axis 0 1 0
                  anchor 0.06 0 0.02
                }
                endPoint Solid {
                  translation 0.06 0 0.02
                  children [
                    DEF r6 Transform {
                      children [
                        USE SMALL_WHEEL_SHAPE
                      ]
                    }
                  ]
                  name "sr6"
                  boundingObject USE r6
                  physics Physics {
                    density -1
                    mass 0.1
                  }
                }
              }
              DEF sr7 HingeJoint {
                jointParameters HingeJointParameters {
                  axis -0.707107 0.707107 0
                  anchor -0.04535 -0.04535 0.02
                }
                endPoint Solid {
                  translation -0.04535 -0.04535 0.02
                  rotation -0.7071067811865476 0.7071067811865476 0 0
                  children [
                    DEF r7 Transform {
                      rotation 0 0 1 0.76
                      children [
                        USE SMALL_WHEEL_SHAPE
                      ]
                    }
                  ]
                  name "sr7"
                  boundingObject USE r7
                  physics Physics {
                    density -1
                    mass 0.1
                  }
                }
              }
              DEF sr8 HingeJoint {
                jointParameters HingeJointParameters {
                  axis -0.707107 0.707107 0
                  anchor 0.04535 0.04535 0.02
                }
                endPoint Solid {
                  translation 0.04535 0.04535 0.02
                  rotation -0.7071067811865476 0.7071067811865476 0 0
                  children [
                    DEF r8 Transform {
                      rotation 0 0 1 0.76
                      children [
                        USE SMALL_WHEEL_SHAPE
                      ]
                    }
                  ]
                  name "sr8"
                  boundingObject USE r8
                  physics Physics {
                    density -1
                    mass 0.1
                  }
                }
              }
              DEF WHEEL_TRANS Transform {
                translation 0 0 0.02
                rotation 1 0 0 1.5708
                children [
                  USE WHEEL_SHAPE
                ]
              }
            ]
            name "solid(1)"
            boundingObject USE WHEEL_TRANS
            physics Physics {
            }
          }
        }
      ]
    }
    DEF WH_3 Transform {
      translation -0.13 0 -0.069
      rotation 0 0.9999999999999999 0 -2.0944
      children [
        DEF WHEEL1 HingeJoint {
          jointParameters HingeJointParameters {
            axis 0 0 1
            anchor 0 0 0.02
          }
          device [
            RotationalMotor {
              name "wheel3"
              maxVelocity 30.3687
              maxTorque 0.8473
            }
            PositionSensor {
              name "ps_3"
              resolution 0.003848
            }
          ]
          endPoint Solid {
            translation 0 0 -0.009999999999999998
            rotation 0 0 1 0
            children [
              DEF sr1 HingeJoint {
                jointParameters HingeJointParameters {
                  anchor 0 0.06 0.02
                }
                endPoint Solid {
                  translation 0 0.06 0.02
                  rotation 1 0 0 0
                  children [
                    DEF r1 Transform {
                      rotation 0 0 1 1.5708
                      children [
                        DEF SMALL_WHEEL_SHAPE Shape {
                          appearance PBRAppearance {
                            baseColor 1 0.752941 0.796078
                            roughness 1
                            metalness 0
                          }
                          geometry Cylinder {
                            height 0.03
                            radius 0.012
                          }
                        }
                      ]
                    }
                  ]
                  name "sr1"
                  boundingObject USE r1
                  physics Physics {
                    density -1
                    mass 0.1
                  }
                }
              }
              DEF sr2 HingeJoint {
                jointParameters HingeJointParameters {
                  axis 0.707107 0.707107 0
                  anchor -0.04535 0.04535 0.02
                }
                endPoint Solid {
                  translation -0.04535 0.04535 0.02
                  rotation 0.7071067811865476 0.7071067811865476 0 0
                  children [
                    DEF r2 Transform {
                      rotation 0 0 1 2.27
                      children [
                        USE SMALL_WHEEL_SHAPE
                      ]
                    }
                  ]
                  name "sr2"
                  boundingObject USE r2
                  physics Physics {
                    density -1
                    mass 0.1
                  }
                }
              }
              DEF sr3 HingeJoint {
                jointParameters HingeJointParameters {
                  anchor 0 -0.06 0.02
                }
                endPoint Solid {
                  translation 0 -0.06 0.02
                  rotation 1 0 0 0
                  children [
                    DEF r3 Transform {
                      rotation 0 0 1 1.5708
                      children [
                        USE SMALL_WHEEL_SHAPE
                      ]
                    }
                  ]
                  name "sr3"
                  boundingObject USE r3
                  physics Physics {
                    density -1
                    mass 0.1
                  }
                }
              }
              DEF sr4 HingeJoint {
                jointParameters HingeJointParameters {
                  axis 0.707107 0.707107 0
                  anchor 0.04535 -0.04535 0.02
                }
                endPoint Solid {
                  translation 0.04535 -0.04535 0.02
                  rotation 0.7071067811865476 0.7071067811865476 0 0
                  children [
                    DEF r4 Transform {
                      rotation 0 0 1 2.27
                      children [
                        USE SMALL_WHEEL_SHAPE
                      ]
                    }
                  ]
                  name "sr4"
                  boundingObject USE r4
                  physics Physics {
                    density -1
                    mass 0.1
                  }
                }
              }
              DEF sr5 HingeJoint {
                jointParameters HingeJointParameters {
                  axis 0 1 0
                  anchor -0.06 0 0.02
                }
                endPoint Solid {
                  translation -0.06 0 0.02
                  children [
                    DEF r5 Transform {
                      children [
                        USE SMALL_WHEEL_SHAPE
                      ]
                    }
                  ]
                  name "sr5"
                  boundingObject USE r5
                  physics Physics {
                    density -1
                    mass 0.1
                  }
                }
              }
              DEF sr6 HingeJoint {
                jointParameters HingeJointParameters {
                  axis 0 1 0
                  anchor 0.06 0 0.02
                }
                endPoint Solid {
                  translation 0.06 0 0.02
                  children [
                    DEF r6 Transform {
                      children [
                        USE SMALL_WHEEL_SHAPE
                      ]
                    }
                  ]
                  name "sr6"
                  boundingObject USE r6
                  physics Physics {
                    density -1
                    mass 0.1
                  }
                }
              }
              DEF sr7 HingeJoint {
                jointParameters HingeJointParameters {
                  axis -0.707107 0.707107 0
                  anchor -0.04535 -0.04535 0.02
                }
                endPoint Solid {
                  translation -0.04535 -0.04535 0.02
                  rotation -0.7071067811865476 0.7071067811865476 0 0
                  children [
                    DEF r7 Transform {
                      rotation 0 0 1 0.76
                      children [
                        USE SMALL_WHEEL_SHAPE
                      ]
                    }
                  ]
                  name "sr7"
                  boundingObject USE r7
                  physics Physics {
                    density -1
                    mass 0.1
                  }
                }
              }
              DEF sr8 HingeJoint {
                jointParameters HingeJointParameters {
                  axis -0.707107 0.707107 0
                  anchor 0.04535 0.04535 0.02
                }
                endPoint Solid {
                  translation 0.04535 0.04535 0.02
                  rotation -0.7071067811865476 0.7071067811865476 0 0
                  children [
                    DEF r8 Transform {
                      rotation 0 0 1 0.76
                      children [
                        USE SMALL_WHEEL_SHAPE
                      ]
                    }
                  ]
                  name "sr8"
                  boundingObject USE r8
                  physics Physics {
                    density -1
                    mass 0.1
                  }
                }
              }
              DEF WHEEL_TRANS Transform {
                translation 0 0 0.02
                rotation 1 0 0 1.5708
                children [
                  USE WHEEL_SHAPE
                ]
              }
            ]
            name "solid(2)"
            boundingObject USE WHEEL_TRANS
            physics Physics {
            }
          }
        }
      ]
    }
    DEF FRONT_BAR Transform {
      translation 0.0779422855494415 0.049999999999999975 0.04500000137032578
      rotation 0 1 0 5.7595865139999995
      children [
        DEF BOT_BOX Shape {
          appearance USE PURPLE
          geometry Box {
            size 0.05 0.1 0.1
          }
        }
        DEF BARRA_10CM Transform {
          translation 0.031243555299131264 0.03999999999999999 2.1316178561853194e-09
          rotation 0 1 0 6.283185307179586
          children [
            Shape {
              appearance PBRAppearance {
              }
              geometry Box {
                size 0.01 0.01 0.1
              }
            }
            DEF FRONT_SENSORS Transform {
              children [
                DEF DS_RIGHT DistanceSensor {
                  translation 0 0 0.055
                  rotation 0 1 0 -0.0872665
                  children [
                    Shape {
                      appearance DEF GREEN PBRAppearance {
                        baseColor 0.45098039215686275 0.8235294117647058 0.08627450980392157
                        metalness 0
                      }
                      geometry DEF BOX1 Box {
                        size 0.01 0.01 0.01
                      }
                    }
                  ]
                  name "ds_right"
                  lookupTable [
                    0 0 0
                    0.2 65535 0
                  ]
                  resolution 1
                }
                DEF DS_LEFT DistanceSensor {
                  translation 0 0 -0.055
                  rotation 0 1 0 0.0872665
                  children [
                    Shape {
                      appearance DEF GREEN PBRAppearance {
                        baseColor 0.45098039215686275 0.8235294117647058 0.08627450980392157
                        metalness 0
                      }
                      geometry DEF BOX1 Box {
                        size 0.01 0.01 0.01
                      }
                    }
                  ]
                  name "ds_left"
                  lookupTable [
                    0 0 0
                    0.2 65535 0
                  ]
                  resolution 1
                }
              ]
            }
          ]
        }
      ]
    }
  ]
  boundingObject USE BODY
  physics Physics {
  }
  controller "u3_exam_munoz_samara"
}
